src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -3929,3 +4077,166 @@ def create_mc(self, mc: ManagedCluster) -> ManagedCluster:
                 else:
                     raise ex
         raise retry_exception
+
+
+class AKSUpdateDecorator:
+    def __init__(
+        self,
+        cmd: AzCliCommand,
+        client: ContainerServiceClient,
+        models: AKSCreateModels,
+        raw_parameters: Dict,
+    ):
+        """Internal controller of aks_update.
+
+        Break down the all-in-one aks_update function into several relatively independent functions (some of them have
+        a certain order dependency) that only focus on a specific profile or process a specific piece of logic.
+        In addition, an overall control function is provided. By calling the aforementioned independent functions one
+        by one, a complete ManagedCluster object is gradually updated and finally requests are sent to update an
+        existing cluster.
+        """
+        self.cmd = cmd
+        self.client = client
+        self.models = models
+        # store the context in the process of assemble the ManagedCluster object
+        self.context = AKSCreateContext(cmd, raw_parameters, mode=DecoratorMode.UPDATE)
+
+    def __record_is_updated(self, is_updated, intermediate_name="updated") -> None:
+        """Helper function to record an indicator of the update status.
+
+        This function will record an indicator indicating the status of the update. The indicator is a boolean
+        intermediate value, once updated, it will be set to True. The purpose is to identify whether the `mc` object
+        has been updated.
+
+        :return: None
+        """
+        previous_state = self.context.get_intermediate(intermediate_name, default_value=False)
+        self.context.set_intermediate(intermediate_name, is_updated or previous_state, overwrite_exists=True)
+
+    def __check_is_updated(self, intermediate_name="updated") -> None:
```

Comment:
could you please clarify more about this function?

Reply:
We have changed the mechanism to check whether the parameters have been set in commit [#123aca0](https://github.com/Azure/azure-cli/pull/19647/commits/123aca085cd4d897e8b2949e53e1c6bdef4f3ad4).